test/reflection_test.cpp

```diff
@@ -373,6 +425,106 @@ TEST(Reflection, FieldCountLimits)
+                                          [&]<class T>(const std::string_view& name, T& field)
+                                          {
+                                              if constexpr (std::is_same_v<int, T>)
+                                              {
+                                                  field = 5;
+                                              }
+
+                                              out.push_back(name);
+                                          });
+
+        return out;
+    }();
+
+    EXPECT_EQ(field_list.at(0), "my_int");
+    EXPECT_EQ(field_list.at(1), "my_double");
+    EXPECT_EQ(field_list.at(2), "my_int2");
+    EXPECT_EQ(field_list.at(3), "my_int_ptr");
+    EXPECT_EQ(field_list.at(4), "int_int_ptr2");
+
+    EXPECT_EQ(a.my_int, 5);
+
+    constexpr GlobalNamespaceStruct8 b{};
+
+    field_list = [&]()
+    {
+        FixedVector<std::string_view, 10> out{};
+        reflection_detail::for_each_field(
+            b, [&]<class T>(const std::string_view& name, const T&) { out.push_back(name); });
+        return out;
+    }();
+
+    EXPECT_EQ(field_list.at(0), "a1");
+    EXPECT_EQ(field_list.at(1), "b2");
+    EXPECT_EQ(field_list.at(2), "c3");
+    EXPECT_EQ(field_list.at(3), "d4");
+    EXPECT_EQ(field_list.at(4), "e5");
+    EXPECT_EQ(field_list.at(5), "f6");
+    EXPECT_EQ(field_list.at(6), "g7");
+    EXPECT_EQ(field_list.at(7), "h8");
+
+    GlobalNamespaceStruct3 c{};
+    field_list = [&]()
+    {
+        FixedVector<std::string_view, 10> out{};
+        reflection_detail::for_each_field(
+            c, [&]<class T>(const std::string_view& name, const T&) { out.push_back(name); });
+        return out;
+    }();
+    EXPECT_EQ(field_list.at(0), "a1");
+    EXPECT_EQ(field_list.at(1), "b2");
+    EXPECT_EQ(field_list.at(2), "c3");
+
+    TemplatedStructWithTwoArgumentsSoItContainsAComma<int, double> d{};
+    field_list = [&]()
+    {
+        FixedVector<std::string_view, 10> out{};
+        reflection_detail::for_each_field(
+            d, [&]<class T>(const std::string_view& name, const T&) { out.push_back(name); });
+        return out;
+    }();
+    EXPECT_EQ(field_list.at(0), "a1");
+}
+
+TEST(Reflection, ForEachField_EmptyStruct)
+{
+    {
+        EmptyStruct empty_struct{};
+        std::size_t counter = 0;
+        [&]()
+        {
+            reflection_detail::for_each_field(
+                empty_struct,
+                [&]<class T>(const std::string_view& /*name*/, const T&) { counter++; });
+        }();
+
+        EXPECT_EQ(0, counter);
+    }
+
+    {
+        EmptyTemplatedStruct<int, double> empty_templated_struct{};
+        std::size_t counter = 0;
+        [&]()
+        {
+            reflection_detail::for_each_field(
+                empty_templated_struct,
+                [&]<class T>(const std::string_view& /*name*/, const T&) { counter++; });
+        }();
+
+        EXPECT_EQ(0, counter);
+    }
+}
+
```

Comment:
We should add tests here for "weird" structs, like those with fields that aren't copyable or moveable, or default constructible. Also we should test `constexpr`ness.

Reply:
Done.